Marlin/Configuration_adv.h

```diff
@@ -36,6 +36,23 @@
   #define THERMAL_PROTECTION_BED_HYSTERESIS 2 // Degrees Celsius
 #endif
 
+/**
+ * When a heater is switched off the temperature still rises for a little time, than cools down.
+ * When a heater is switched on the temperature may fall for a little while until the heat reaches the thermometer - than rises.
+ * MAX_TEMP_OVESTOOT_TIME is the longest time a system will need to reach the temperature again, it had when switched On/Off ( in seconds )
+ * This is used to check if a heater is really switched On/Off and the temperature reacts accordingly.
+ * OVERSTOOT is usually much longer than UNDERSHOOT - hence the name here.
+ * Set to 0 to deactivate this test
+ * Because when the heater is Off the temperature can not fall below the ambient temperature, what can differ, we need a limit.
+ * Hence we don't check for falling temperatures below MAX_AMBIENT_TEMP (in degree Celsius).
+ * Default is 50°C. If you have a heat-chamber you probably have to rise this.
+ * To not produce false positives we need some margin for noise at the ADCs
+ * MAX_SENSOR_NOISE_C is the maximum difference of temperature readings when real temperature does not change (in degree Celsius).
+ */
+#define MAX_TEMP_OVERSTOOT_TIME 180
```

Comment:
I don't know why a J-head would need 3 minutes for the temperature to be
moving in the right direction. I anticipated a few seconds and the entire
heating process for my J-heads is less than 3 minutes.

On 17 May 2015 at 02:59, AnHardt <notifications@github.com> wrote:

> In Marlin/Configuration_adv.h
> <https://github.com/MarlinFirmware/Marlin/pull/2092#discussion_r30466623>:
>
> > @@ -36,6 +36,23 @@
> >    #define THERMAL_PROTECTION_BED_HYSTERESIS 2 // Degrees Celsius
> >  #endif
> >
> > +/**
> > + * When a heater is switched off the temperature still rises for a little time, than cools down.
> > + * When a heater is switched on the temperature may fall for a little while until the heat reaches the thermometer - than rises.
> > + * MAX_TEMP_OVESTOOT_TIME is the longest time a system will need to reach the temperature again, it had when switched On/Off ( in seconds )
> > + * This is used to check if a heater is really switched On/Off and the temperature reacts accordingly.
> > + * OVERSTOOT is usually much longer than UNDERSHOOT - hence the name here.
> > + * Set to 0 to deactivate this test
> > + * Because when the heater is Off the temperature can not fall below the ambient temperature, what can differ, we need a limit.
> > + * Hence we don't check for falling temperatures below MAX_AMBIENT_TEMP (in degree Celsius).
> > + * Default is 50°C. If you have a heat-chamber you probably have to rise this.
> > + * To not produce false positives we need some margin for noise at the ADCs
> > + * MAX_SENSOR_NOISE_C is the maximum difference of temperature readings when real temperature does not change (in degree Celsius).
> > + */
> > +#define MAX_TEMP_OVERSTOOT_TIME 180
>
> @thinkyhead <https://github.com/thinkyhead>
> To be exact i mean OVERSHOOT.
> It's always the same with me. I write it once, and than allays copy, to be
> sure it's the same. So spelling does not matter for the function - but has
> to be correct the first time, to be the right English word.
> Will change this later.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/MarlinFirmware/Marlin/pull/2092/files#r30466623>.
>